tests/contrib/structlog/test_structlog_logging.py

```diff
@@ -0,0 +1,295 @@
+import json
+
+import pytest
+import structlog
+
+from ddtrace import config
+from ddtrace.constants import ENV_KEY
+from ddtrace.constants import SERVICE_KEY
+from ddtrace.constants import VERSION_KEY
+from ddtrace.contrib.structlog import patch
+from ddtrace.contrib.structlog import unpatch
+from tests.utils import DummyTracer
+
+
+cf = structlog.testing.CapturingLoggerFactory()
+
+
+def _test_logging(output, span, env="", service="", version=""):
+    dd_trace_id, dd_span_id = (span.trace_id, span.span_id) if span else (0, 0)
+
+    assert json.loads(output[0].args[0])["event"] == "Hello!"
+    assert json.loads(output[0].args[0])["dd.trace_id"] == str(dd_trace_id)
+    assert json.loads(output[0].args[0])["dd.span_id"] == str(dd_span_id)
+    assert json.loads(output[0].args[0])["dd.env"] == env or ""
+    assert json.loads(output[0].args[0])["dd.service"] == service or ""
+    assert json.loads(output[0].args[0])["dd.version"] == version or ""
+
+    cf.logger.calls.clear()
+
+
+@pytest.fixture(autouse=True)
+def patch_structlog():
+    patch()
+    structlog.configure(
+        processors=[structlog.processors.JSONRenderer()],
+        logger_factory=cf,
+    )
+    yield
+    unpatch()
+
+
+@pytest.fixture(autouse=True)
+def global_config():
+    config.service = "logging"
+    config.env = "global.env"
+    config.version = "global.version"
+    yield
+    config.service = config.env = config.version = None
+
+
+def test_log_trace_global_values():
+    """
+    Check trace info includes global values over local span values
+    """
+
+    tracer = DummyTracer()
+    span = tracer.trace("test.logging")
+    span.set_tag(ENV_KEY, "local-env")
+    span.set_tag(SERVICE_KEY, "local-service")
+    span.set_tag(VERSION_KEY, "local-version")
+
+    structlog.get_logger().info("Hello!")
+    if span:
+        span.finish()
```

Comment:
Span should be None in this test
```suggestion
    span.finish()
```